src/main/java/wellnus/common/MainManager.java

```diff
@@ -48,37 +53,61 @@ private void executeCommands() {
             try {
                 String nextCommand = this.getTextUi().getCommand();
                 String featureKeyword = parser.getMainArgument(nextCommand);
-                HashMap<String, String> arguments = parser.parseUserInput(nextCommand);
                 Optional<Manager> featureManager = this.getManagerFor(featureKeyword);
                 if (featureManager.isEmpty() && !this.isSupportedCommand(featureKeyword)) {
                     BadCommandException badCommandException =
                             new BadCommandException(MainManager.INVALID_COMMAND_MESSAGE);
                     this.getTextUi().printErrorFor(badCommandException,
                             MainManager.INVALID_COMMAND_ADDITIONAL_MESSAGE);
+                    continue;
                 }
-                // TODO: Replace with Command subclass once those changes are merged
-                if (featureKeyword.equals("exit")) {
-                    isExit = true;
-                }
+                featureManager.ifPresent((manager) -> {
+                    // TODO: Consider if there's a way to avoid this extra try-catch?
+                    try {
+                        manager.runEventDriver();
+                    } catch (BadCommandException badCommandException) {
+                        this.getTextUi().printErrorFor(badCommandException, NO_ADDITIONAL_MESSAGE);
+                    }
```

Comment:
Can I check why this try catch is needed to run the method runEventDriver? I do not see the method throwing an exception